spec/acceptance/helpers.rb

```diff
@@ -0,0 +1,75 @@
+# Time constants in seconds
+HOUR =  60 * 60
+DAY  =  24 * HOUR
+YEAR = 365 * DAY
+
+# Execute `date` command on host with optional arguments
+# and get back a Ruby Time object
+#
+# @param [Host]            host   single Beaker::Host to run the command on
+# @param [Array<String>]   args   Array of arguments to be appended to the
+#                                 `date` command
+# @return [Time]    Ruby Time object
+def get_time_on(host, args = [])
+  arg_string = args.join(' ')
+  date = on(host, "date #{arg_string}").stdout.chomp
+  return Time.parse(date)
+end
+
+# Retrieve the CA enddate on a given host as a Ruby time object
+#
+# @param [Host]            host   single Beaker::Host to get CA enddate from
+#
+# @return [Time]    Ruby Time object
+def get_ca_enddate_time_on(host)
+  if host[:roles].include? 'master' then
+    ca_path = '/etc/puppetlabs/puppet/ssl/ca/ca_crt.pem'
+  else
+    ca_path = '/etc/puppetlabs/puppet/ssl/certs/ca.pem'
+  end
+  on(host, "openssl x509 -in #{ca_path} -noout -enddate") do |result|
+    m = result.stdout.match(/notAfter=(.*) (\d{4}) ([A-Z]{3})$/)
+    if m then
+      enddate = "#{m[1]} #{m[3]} #{m[2]}"
+      return Time.parse(enddate)
+    end
+  end
+  return Time.at(0)
+end
+
+# Retrieve the current ca_serial value for `puppet certgen ca` on a given host
+#
+# @param [Host]            host   single Beaker::Host to get ca_serial from
+#
+# @return [String]    ca_serial in hexadecimal, 'FFFFFFFF' if error
+def get_ca_serial_id_on(host)
+  err = on(host, puppet("certregen ca"), :acceptable_exit_codes => [0,1]).stderr
+  m = err.match(/rerun this command with --ca_serial ([0-9a-fA-F]+)/)
+  if m then
+    return m[1]
+  end
+  return 'FFFFFFFF'
```

Comment:
If we can't determine the CA serial number then using an arbitrarily large number will still fail but later on. Can this just throw an error or cause a test failure?